### PR TITLE
fix: add exports main field

### LINKED ISSUE #433: ESM declared as `module` main field instead of `exports`
## Expected Behavior
When using ESM with `import { attributesToProps } from 'html-react-parser';` it should import the `index.mjs` entry, and `attributesToProps` should be a function.

## Actual Behavior

It instead tries to import `index.js` and fails with:
```
SyntaxError: Named export 'attributesToProps' not found. The requested module 'html-react-parser' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'html-react-parser';
const { attributesToProps } = pkg;
```

## Potential Solution
The package currently uses `"module": "index.mjs"`, the ESM spec has since use `"exports": "./index.mjs"`. So the solution would be to swap `module` for `exports`, or leave `module` in I suppose for backwards compatibility?

## Steps to Reproduce

- Create a ESM project e.g: `"type": "module` in the `package.json`.
- Install `html-react-parser`.
- Create a file along the lines of `import { attributesToProps } from 'html-react-parser';`

## Reproducible Demo

Not 100% if those platforms support ESM.

## Environment

- Version: 1.2.7
- Platform: macOS Big Sure 11.2.3
- Browser: Chromium 97

### PR DESCRIPTION
**Pull request by AndrewLeedham**
Fixes #433

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "author": "Mark <mark@remarkablemark.org>",
   "main": "index.js",
   "module": "index.mjs",
+  "exports": "./index.mjs",
```
Comment (remarkablemark) [inline package.json:8]: Would this cause a regression for those who use CommonJS?

I see in other [areas](https://stackoverflow.com/questions/58990498/new-package-json-exports-field-not-working-with-typescript) that `import`, `require`, and `types` are defined in `exports`:

```json
{
  "exports": {
    ".": {
      "import": "./index.mjs",
      "require": "./index.js",
      "types": "./index.d.ts"
    }
  }
}
```

Comment (AndrewLeedham) [reply package.json]: Nice, hadn't come across that syntax. Updated the PR 😃

Comment (remarkablemark) [reply package.json]: Thanks!

**Review by remarkablemark — APPROVED** [review]

**Comment by remarkablemark** [comment]
Thanks for the contribution @AndrewLeedham. You can find the release here: https://github.com/remarkablemark/html-react-parser/pull/436#issuecomment-1024614229

**Pull request merged**